validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -69,10 +69,7 @@ def __init__(self, state_hash, read_list, write_list, base_context_ids):
         self._state = {}
         self.base_context_ids = base_context_ids
 
-        self._id = hashlib.sha256((str(state_hash) + ":" +
-                                  str(read_list + write_list) + ":" +
-                                  time.time().hex()).encode()
-                                  ).hexdigest()
+        self._id = hashlib.sha256(uuid.uuid4().hex.encode()).hexdigest()
```

Comment:
Well, surely if you get a collision on uuid.uuid4(), hashing the result is not going to help :)